autorevision.sh

```diff
@@ -1335,8 +1339,20 @@ if [ -f "${CACHEFILE}" ] && [ "${CACHEFORCE}" = "1" ]; then
 else
 	# If a value is not set through the environment set VCS_EXTRA to nothing.
 	: "${VCS_EXTRA:=""}"
+
+	if [ ! -z "${CHANGE_TO_DIR}" ]; then
+		originalPath="${PWD}"
```

Comment:
Why not 
```suggestion
		originalPath="$(pwd)"
```
instead since `${PWD}` could be overwritten with an unexpected value?

Reply:
I think you might be right. I did it this way because it is done like this in various other places in the script (such as `${LOCAL} oldPath="${PWD}"`). There are more than 10 occurrences of `${PWD}`, but `$(pwd)` isn't used at all.

Then again:
> If an application sets or unsets the value of PWD, the behavior of pwd is unspecified.

(from http://pubs.opengroup.org/onlinepubs/9699919799/utilities/pwd.html)

How about keeping it with `${PWD}` for this patch, and change all occurrences of `${PWD}` to `$(pwd)` in another patch?

Reply:
> I think you might be right. I did it this way because it is done like this in various other places in the script (such as `${LOCAL} oldPath="${PWD}"`). There are more than 10 occurrences of `${PWD}`, but `$(pwd)` isn't used at all.

That sounds good to me; would you like to add that to this pull request?

Reply:
Yes, I'll do that tomorrow.

Reply:
It's done.

I have noticed that the test.sh script fails (already failed on master): `mktemp: too few X's in template ‘autorevision’`
Changing `mktemp -dqt autorevision` into something like `mktemp -dqt autorevision.XXXXXX` makes it work. Want me to add this change as well?

Reply:
> It's done.
> 
> I have noticed that the test.sh script fails (already failed on master): `mktemp: too few X's in template ‘autorevision’`
> Changing `mktemp -dqt autorevision` into something like `mktemp -dqt autorevision.XXXXXX` makes it work. Want me to add this change as well?

Unlike autorevision its self the test script is in no way designed to be portable and is only intended to work on a mac (really only on my machine); this is largely due to the fact that the 'testing' it does is of strictly limited utility. So if you want to make something that does more comprehensive testing I would be all for it but as things currently stand I would rather not have that change put in by its self.

Reply:
OK, never mind then. I thought test.sh was in a kind of semi-abandoned state (last change in 2016) in need of a little fix. So, let's just keep it the way it is.